app/repositories/page-content.tsx

```diff
@@ -4,22 +4,79 @@ import { useSearchParams } from 'next/navigation'
 import { useInfiniteQuery } from '@tanstack/react-query'
 import client from '@/lib/client'
 import { useTitle } from 'react-use'
+import { ToolsOzoneModerationDefs } from '@atproto/api'
+
+const isEmailSearch = (q: string) => q.startsWith('email:')
+
+const getSearchResults = async ({
+  q,
+  cursor,
+}: {
+  q: string
+  cursor?: string
+}): Promise<{
+  cursor?: string
+  repos: ToolsOzoneModerationDefs.RepoView[]
+}> => {
+  const headers = { headers: client.proxyHeaders() }
+  const limit = 25
+
+  if (!isEmailSearch(q)) {
+    const { data } = await client.api.tools.ozone.moderation.searchRepos(
+      { q, limit, cursor },
+      headers,
+    )
+
+    return data
+  }
+
+  const email = q.replace('email:', '').trim()
+
+  if (!email) {
+    return { repos: [], cursor: undefined }
+  }
+
+  const { data } = await client.api.com.atproto.admin.searchAccounts(
+    { email, limit, cursor },
+    headers,
+  )
+
+  if (!data.accounts.length) {
+    return { repos: [], cursor: data.cursor }
+  }
+
+  const repos: Record<string, ToolsOzoneModerationDefs.RepoView> = {}
+  data.accounts.forEach((account) => {
+    repos[account.did] = {
+      ...account,
+      // Set placeholder properties that will be later filled in with data from ozone
+      relatedRecords: [],
+      indexedAt: account.indexedAt,
+      moderation: {},
+      labels: [],
+    }
+  })
+
+  await Promise.allSettled(
+    data.accounts.map(async (account) => {
```

Comment:
Without handling the result of `allSettled()` this will swallow errors, including unexpected ones.  Another option is to use `Promise.all() here` and handle expected error conditions within each item.

Reply:
This is intentional, I think we have most info we need to render a partial view from the resultSet so if some requests fail, they will just not show the augmented data from getRepo()

Reply:
That totally makes sense and I don't think we'd need to change that— my main question is whether we want to swallow all errors versus ignore some acceptable set of errors and log or throw the rest.  For example, a logic bug could cause a `TypeError` in here and there's no way we'd be able to observe it, could easily go unnoticed.

Reply:
The rule here might be that we're comfy ignoring all `XRPCErrors`, for example.  I really appreciate libraries like [bounce](https://github.com/hapijs/bounce/blob/master/API.md) (though I think this may catered to server-side) which ensure developer/system errors don't get missed when permitting some errors.